server-mirror-git/src/main/java/com/linecorp/centraldogma/server/internal/mirror/GitWithAuth.java

```diff
@@ -187,28 +211,59 @@ public LsRemoteCommand lsRemote() {
 
     private static <T extends TransportCommand<?, ?>> void configureSsh(T cmd, PublicKeyMirrorCredential cred) {
         cmd.setTransportConfigCallback(transport -> {
-            final SshTransport sshTransport = (SshTransport) transport;
-            final JschConfigSessionFactory sessionFactory = new JschConfigSessionFactory() {
+            final Collection<KeyPair> keyPairs;
+            try {
+                keyPairs = keyPairResourceParser.loadKeyPairs(null, NamedResource.ofName(cred.username()),
+                                                              passwordProvider(cred), cred.privateKey());
+            } catch (IOException | GeneralSecurityException e) {
+                throw new MirrorException("Unexpected exception while loading private key. username: " +
+                                          cred.username() + ", publicKey: " +
+                                          publicKeyPreview(cred.publicKey()), e);
+            }
```

Comment:
Optional) Would be better to eagerly create `keyPairs` and raise an exception if invalid before calling `cmd.setTransportConfigCallback`?

Reply:
That's a good suggestion. Fixed. 👍